src/utils/store/customizeInstanceType.ts

```diff
@@ -0,0 +1,38 @@
+import produce from 'immer';
+
+import { V1VirtualMachine } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { isEmpty } from '@kubevirt-utils/utils/utils';
+import { effect, signal } from '@preact/signals-react';
+
+import {
+  getCustomizeInstanceTypeSessionStorage,
+  saveCustomizeInstanceTypeSessionStorage,
+} from './customizeInstanceType/utils/utils';
+
+export const vmSignal = signal<V1VirtualMachine>(null);
```

Comment:
In the comments of the video you recommended we watch I saw warnings about using signals because it patches React internals. I saw one comment that said that a second blog post from Preact noted that they are no longer doing that, but I didn't see that when I read the second blog post. Can we be absolutely sure that signals won't cause unexpected problems down the line?

Reply:
This is a great question! 

You are correct that older versions changed the internal. While this is fine on current React versions as it's thoroughly tested and works with them (for regular React setup), it may cause issues with the next React releases.
But as you can see in signal-react, releases https://github.com/preactjs/signals/releases/tag/%40preact%2Fsignals-react%402.0.0, they are removing the internal changes because they had issues that were hard to debug, and values weren't updated (mostly in next. js—from reading posts). By the way, we are using 2.0.1. 

Anyway, if any bugs surface while using signals, there is only one single file for the store that is consumed, it will be easy to replace with any other store mechanism if we encounter any problems. It's very modular, and there is a separation between logic and rendering.